pkg/controllers/management/kontainerdrivermetadata/data.go

```diff
@@ -387,97 +400,252 @@ func getWindowsName(str string) string {
 	return fmt.Sprintf("w%s", str)
 }
 
-func updateSettings(maxVersionForMajorK8sVersion map[string]string, rancherVersion string,
-	K8sVersionServiceOptions map[string]v3.KubernetesServicesOptions, DefaultK8sVersions map[string]string, deprecated map[string]bool) error {
+func (md *MetadataController) updateSettings(maxVersionForMajorK8sVersion map[string]string, rancherVersion string,
+	K8sVersionServiceOptions map[string]v3.KubernetesServicesOptions, DefaultK8sVersions map[string]string,
+	deprecated map[string]bool) error {
+
+	userSettings, userUpdated, err := md.getUserSettings()
+	if err != nil {
+		return err
+	}
+
+	updateSettings, err := toUpdate(maxVersionForMajorK8sVersion, deprecated, DefaultK8sVersions, rancherVersion, K8sVersionServiceOptions)
+	if err != nil {
+		return err
+	}
+
+	if !userUpdated {
+		if err := md.updateSettingFromFields(updateSettings, false); err != nil {
+			return err
+		}
+	} else {
+		userMaxVersionForMajorK8sVersion, userDeprecated, err := getUserSettings(userSettings, DefaultK8sVersions)
+		if err != nil {
+			return err
+		}
+
+		if len(userMaxVersionForMajorK8sVersion) == 0 {
+			userMaxVersionForMajorK8sVersion = maxVersionForMajorK8sVersion
+		}
+
+		if len(userDeprecated) == 0 {
+			userDeprecated = deprecated
+		}
+
+		userUpdateSettings, err := toUpdate(userMaxVersionForMajorK8sVersion, userDeprecated, DefaultK8sVersions, rancherVersion, K8sVersionServiceOptions)
+		if err != nil {
+			return err
+		}
+
+		if err := md.updateSettingFromFields(userUpdateSettings, false); err != nil {
+			return err
+		}
+	}
+
+	if err := md.updateSettingFromFields(updateSettings, true); err != nil {
```

Comment:
Guess this call is to make sure defaults of all settings is updated. Do we need to update the `updateSettings` here?

Reply:
@kinarashah does it make sense to update all default values at the start of this routine and then update the setting values? Wondering if that will be good to have defaults uptodate,  in case if we hit some error updating setting values.

Reply:
don't think the order would matter either way, because we always read from value and not from default, but updated